### ISSUE TITLE
Allow the option to confirm the password and make the encrypted line not removable

### ISSUE THREAD

**Issue by The-Eighth-Man** [meld-cp/obsidian-encrypt#3]
As we only set the password once, and there is no option to see the password, one single mistake when setting it and we can't ever access the file again, thus a second reentering of the password will help to mitigate this problem. 

The encryption is good but it matters little if it can be easily deleted (by accident or someone else), thus I propose to make the line not removable by defect or by an additional command.

**Comment by jtmusky** [comment]
Password validation is a good idea. As I just accidentally used the <Esc> key as my password. ha

**Comment by meld-cp** [comment]
> Password validation is a good idea. As I just accidentally used the **Esc** key as my password. ha

😬 Whoops, sorry about that. Yeah, let me look into it. 👍

**Comment by meld-cp** [comment]
I've created a new issue for the ESC bug (#5)

This issue will be used to track the confirm password enhancement.

**Comment by meld-cp** [comment]
I've added a confirm password box when encrypting in release 1.3.0.

Sorry @The-Eighth-Man , I don't have a way to make the line not removable or read-only after it has been encrypted.

**Comment by mayurankv** [comment]
Sorry if this is drudging open an old topic, a better idea than comments could be embedding a new attachment file with the encrypted string in it (possibly also in a comment). (i.e. this new file would go into whatever attachment folder a user has set).
The benefits are:
- **Much less likely to accidentally edit the encrypted string** <- The main one
- Current mechanisms still work with this idea:
	- It would still be clear what needs to be decrypted (the only text in the new file)
	- It would still be able to decrypt in place (just replace the embed)
- Could put hint in file name 
- This wouldn't even need to be a .md file, it could be any file format including dotfiles making the encrypted strings more hidden if cluttering was an issue

Possible issues:
- Naming conflicts?: Shouldn't be an issue if timestamped (security concerns about stamping idk?)
- Respecting linking format of user?: Also I don't think should be a problem.

Ideally this would be optional but I do not know how hard/easy this would be in the first place, let alone as an option.

Please let me know if this would be possible/ if you have any questions at all! :)

**Comment by meld-cp** [comment]
Hi @mayurankv, totally agree.

I've been working on having a separate encrypted file.

Although progress is slow, I have a proof of concept working since last weekend.

I just need to tidy up a few things.

It will also help with issues #31 and #32

**Comment by mayurankv** [comment]
Great to hear! Really looking forward to it and thank you for the update.

One last suggestion I have would be automatically backing up any encrypted string accompanied with a time stamp to some chosen backup file so even if a file is changed or deleted, there is still this master backup of encrypted strings that it can be recovered from. Again just floating an idea but thank you so much for this plugin!

**Comment by meld-cp** [comment]
@mayurankv hopefully v2.0.0 is a better workflow. 🥳
